l10n_ar_ux/controllers/portal.py

```diff
@@ -34,22 +34,35 @@ def details_form_validate(self, data):
 
     @route()
     def account(self, redirect=None, **post):
-        if post:
-            error, _error_message = self.details_form_validate(post)
-            if not error:
-                post.pop('commercial_partner_id', False)
-                post.pop('vat', False)
-                post.pop('l10n_latam_identification_type_id', False)
-                post.pop('l10n_ar_afip_responsibility_type_id', False)
+        # if post:
+        #     error, _error_message = self.details_form_validate(post)
+        #     if not error:
+        #         post.pop('commercial_partner_id', False)
+        #         post.pop('vat', False)
+        #         post.pop('l10n_latam_identification_type_id', False)
+        #         post.pop('l10n_ar_afip_responsibility_type_id', False)
 
         response = super().account(redirect=redirect, **post)
         identification_types = request.env['l10n_latam.identification.type'].sudo().search([])
         afip_responsibilities = request.env['l10n_ar.afip.responsibility.type'].sudo().search([])
         uid = request.session.uid
         partner = request.env['res.users'].browse(uid).partner_id if uid else request.env['res.partner']
         partner = partner.with_context(show_address=1).sudo()
+        #recomputamos el commercial partner porque al editar el fomrulario perdemos ese dato
+        partner._compute_commercial_partner()
+        #Al guardar por primera vez los datos del contacto, se pierden la responsabilidad y el tipo de identificacion
+        # if not partner.l10n_ar_afip_responsibility_type_id and post.get('l10n_ar_afip_responsibility_type_id'):
+        #     resp = int(post.get('l10n_ar_afip_responsibility_type_id'))
+        #     partner.l10n_ar_afip_responsibility_type_id = request.env['l10n_ar.afip.responsibility.type'].browse(resp)
+        # if not partner.l10n_latam_identification_type_id and post.get('l10n_latam_identification_type_id'):
+        #     ident = int(post.get('l10n_latam_identification_type_id'))
+        #     partner.l10n_latam_identification_type_id = request.env['l10n_latam.identification.type'].browse(ident)
+
         response.qcontext.update({
+            'l10n_ar_afip_responsibility_type_id': post.get('l10n_ar_afip_responsibility_type_id'),
+            'l10n_latam_identification_type_id': post.get('l10n_latam_identification_type_id'),
```

Comment:
cambiarle el nombre a uno distinto al del campo
para que asi podamos idenficiar entre el input del form, y el valor real
```suggestion
            'afip_responsibility': post.get('l10n_ar_afip_responsibility_type_id'),
            'lidentification_type': post.get('l10n_latam_identification_type_id'),
```